### PR TITLE
Add worldtube excision sphere tag

### PR DESCRIPTION
**Pull request by nikwit**
adds a tag that returns the radius of the worldtube

~~depends on #4654~~

### CODE REVIEW

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp [hunk 1]
```diff
@@ -3,12 +3,54 @@
...
+  static constexpr Options::String help{
+      "The name of the excision sphere as returned by the domain."};
+  using group = Worldtube;
+};
+
 }  // namespace OptionTags
+
+namespace Tags {
+template <size_t Dim>
+struct WorldtubeRadius : db::SimpleTag {
```
Comment (nilsvu) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp]: The excision sphere essentially _is_ your worldtube, right? So how about you name this `Worldtube` and make it hold the excision sphere. Then you can get the radius, center, etc from it whenever you want. Right now you can just copy the excision sphere from the domain, but we should get #3142 merged so you can make this a reference tag.

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp [hunk 2]
```diff
@@ -3,12 +3,54 @@
...
+  using type = double;
+  using option_tags = tmpl::list<domain::OptionTags::DomainCreator<Dim>,
+                                 OptionTags::ExcisionSphere>;
+  static constexpr bool pass_metavariables = false;
+  static double create_from_options(
+      const std::unique_ptr<::DomainCreator<Dim>>& domain_creator,
```
Comment (nilsvu) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp:55]: I think the argument type should be `const DomainCreator<Dim>&`. This is easy to test with `TestHelpers::test_option_tag_factory_creation`. Please use that in your test.

**Review by nilsvu — CHANGES_REQUESTED** [review]
Please add dox to things so they show up in Doxygen

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp (hunk 2 above)
Comment (nikwit) [reply src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp:55]: no, I think it's a pointer. I don't understand, `TestHelpers::test_option_tag_factory_creation` tests option tags with the type of a base class pointer which I do not have here.

Comment (nilsvu) [reply src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp:55]: You're right. I don't remember if we have a quick way to test an option-created tag like this. The way you did seems fine.

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ */
+namespace OptionTags {
+/*!
+ * \brief Options for the worldtube
+ */
+struct Worldtube {
+  static constexpr Options::String help = {"Options for the Worldtube"};
+};
+/*!
+ * \brief Name of the excision sphere desginated to act as a worldtube
```
Comment (nilsvu) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp]: Typo "designated"

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+/*!
+ * \brief Tags related to the worldtube
+ */
+namespace Tags {
+/*!
+ * \brief The excision sphere corresponding to the worldtube
+ */
+template <size_t Dim>
+struct ExcisionSphere : db::SimpleTag {
```
Comment (nilsvu) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp:49]: Do you prefer this name over `Worldtube`? Up to you

**Review by nilsvu — DISMISSED** [review]
You can fix the typo when rebasing

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp (hunk 4 above)
Comment (nikwit) [reply src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp:49]: yes, I think so. The worldtube has a lot more properties than the `ExcisionSphere` which are mostly in the databox of the Singleton which I will add.

**Review by nilsvu — APPROVED** [review]

**Pull request merged**